### PR TITLE
Ddf 1916- Integrate Search UI 'help' page into documentation

### PR DESCRIPTION
**Pull request by ricklarsen**
#### What does this PR do?
Integrates Search UI 'Help' end user documentation into complete documentation.

#### Who is reviewing it?
@jrnorth @mcalcote @Lambeaux @oscaritoro @vinamartin @stustison 
 
#### How should this be tested?
Successful build, no 'Unresolved Directive' error messages in compiled documentation, no broken image links (see below for exceptions)

#### Any background context you want to provide?
Some images are missing from the Help/Using section. See [DDF-2013](https://codice.atlassian.net/browse/DDF-2013)

#### What are the relevant tickets?
[DDF-1916](https://codice.atlassian.net/browse/DDF-1916) 
#### Checklist:
- [x] Documentation Updated
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

### CODE REVIEW

File: catalog/ui/search-ui/standard/src/assembly/asciidocs.xml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    </formats>
+    <fileSets>
+        <fileSet>
+             <fileMode>0644</fileMode>
+             <directoryMode>0755</directoryMode>
+             <directory>src/main/doc/_contents</directory>
+             <includes>
+                   <include>help-contents.adoc</include>
+            </includes>
+            <outputDirectory></outputDirectory>
```
Comment (vinamartin) [inline catalog/ui/search-ui/standard/src/assembly/asciidocs.xml]: Is this meant to be empty?

File: catalog/ui/search-ui/standard/src/assembly/images.xml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    </formats>
+    <fileSets>
+        <fileSet>
+             <fileMode>0644</fileMode>
+             <directoryMode>0755</directoryMode>
+             <directory>src/main/webapp/images</directory>
+             <includes>
+                   <include>**</include>
+            </includes>
+            <outputDirectory></outputDirectory>
```
Comment (vinamartin) [inline catalog/ui/search-ui/standard/src/assembly/images.xml]: Same question as above

Comment (oscaritoro) [inline catalog/ui/search-ui/standard/src/assembly/images.xml]: Did you truly mean to check in files onto a versioned folder? (ie. ` distribution/docs/images/standard-2.9.0-SNAPSHOT/...`)

File: catalog/ui/search-ui/standard/pom.xml [hunk 3]
```diff
@@ -214,9 +214,16 @@
                 <groupId>org.asciidoctor</groupId>
                 <artifactId>asciidoctor-maven-plugin</artifactId>
                 <version>1.5.2</version>
+                <dependencies>
+                    <dependency>
+                        <groupId>org.asciidoctor</groupId>
+                        <artifactId>asciidoctorj-pdf</artifactId>
+                        <version>1.5.0-alpha.6</version>
```
Comment (shaundmorris) [inline catalog/ui/search-ui/standard/pom.xml]: Why do we need an `alpha` version of this dependency? Seems like there is a 1.5.4 released: https://github.com/asciidoctor/asciidoctorj/releases/tag/v1.5.4 or am I missing something?

File: catalog/ui/search-ui/standard/src/assembly/asciidocs.xml (hunk 1 above)
Comment (ricklarsen) [inline catalog/ui/search-ui/standard/src/assembly/asciidocs.xml]: Yes. It lets the plugin that downloads it decide which directory the images should end up in.

File: catalog/ui/search-ui/standard/pom.xml (hunk 3 above)
Comment (ricklarsen) [inline catalog/ui/search-ui/standard/pom.xml]: I haven't been on the asciidoctor-pdf docs in a while, but as I understand, it's still technically "experimental."

File: catalog/ui/search-ui/standard/src/assembly/images.xml (hunk 2 above)
Comment (ricklarsen) [inline catalog/ui/search-ui/standard/src/assembly/images.xml]: fixed.

Comment (oscaritoro) [inline catalog/ui/search-ui/standard/src/assembly/images.xml]: Thanks for fixing it & sorry for commenting in the wrong place! I meant to comment on the overall PR, but accidentally commented under Vina's comment for an unrelated file.

File: catalog/ui/search-ui/standard/pom.xml (hunk 3 above)
Comment (oscaritoro) [inline catalog/ui/search-ui/standard/pom.xml]: In the dib repo we have root pom level version variables that we use for the plugin and the asciidoctor dependencies... In ddf we only have the one inside the distribution/docs/pom.xml... should we create a ticket to have these at the ddf root pom level?

Comment (oscaritoro) [inline catalog/ui/search-ui/standard/pom.xml]: We have variables inside the distribution/docs/pom.xml... should we create a ticket to have these at the ddf root pom level?

Comment (ricklarsen) [inline catalog/ui/search-ui/standard/pom.xml]: Probably

File: catalog/ui/search-ui/standard/src/main/doc/_contents/help-contents.adoc [hunk 4]
```diff
@@ -0,0 +1,373 @@
...
+
+|Item
+|Description
+
+|Source
+|List of all sources searched with check boxes to allow users to refine searches to the most
+ relevant sources.
+
+|Metadata Content Type
+|List of metadata content types found in the search, with the ability to select and deselect
```
Comment (Lambeaux) [inline catalog/ui/search-ui/standard/src/main/doc/_contents/help-contents.adoc]: Technically this may not be worth changing, but useful FYI: no comma is needed here: `...in the search, with the ability...`; we may take a natural pause when speaking this but it's not a run-on sentence. Once again, mainly pointing out for future reference. I also could be wrong.

File: catalog/ui/search-ui/standard/src/main/doc/_contents/help-contents.adoc [hunk 5]
```diff
@@ -0,0 +1,373 @@
...
+ content type.
+
+|Query
+|The Query builder enables users to construct a very granular search query.
+ The first drop down menu contains the metadata elements in the search results, and the second
```
Comment (Lambeaux) [inline catalog/ui/search-ui/standard/src/main/doc/_contents/help-contents.adoc]: Also not sure we need the comma here: `...in the search results, and the second contains...`

File: catalog/ui/search-ui/standard/src/main/doc/_contents/help-contents.adoc [hunk 6]
```diff
@@ -0,0 +1,373 @@
...
+ contains operators based on the field selected (greater than, equals, contains, matchcase, before, after,
+ etc.)  Click the + to add further constraints to the query, or x to remove.
```
Comment (Lambeaux) [inline catalog/ui/search-ui/standard/src/main/doc/_contents/help-contents.adoc]: And here: `...to the query, or x to remove.`

File: catalog/ui/search-ui/standard/src/main/doc/_contents/help-contents.adoc [hunk 7]
```diff
@@ -0,0 +1,373 @@
...
+ Click Search to use the new query.
+
+|Search button
+|Executes enhanced search on new parameters specified or query built above.
```
Comment (Lambeaux) [inline catalog/ui/search-ui/standard/src/main/doc/_contents/help-contents.adoc]: This reads better: `Executes an enhanced search on any new parameters that were specified or the query built above.`

File: catalog/ui/search-ui/standard/src/main/doc/_contents/help-contents.adoc [hunk 8]
```diff
@@ -0,0 +1,373 @@
...
+. Select the Save button.
+
+
+[NOTE]
+====
+The size of the product is based on the value in the associated metacard's resource-size attribute.
+This is defined when the metacard was originally created and may or may not be accurate. Often it
+will be set to N/A, indicating that the size is unknown or not applicable.
+
+However, if the administrator has enabled caching on {branding} and has installed the
```
Comment (Lambeaux) [inline catalog/ui/search-ui/standard/src/main/doc/_contents/help-contents.adoc]: This is a written list. Modify to this: `...caching on {branding}, has installed the...`

File: catalog/ui/search-ui/standard/src/main/doc/_contents/help-contents.adoc [hunk 9]
```diff
@@ -0,0 +1,373 @@
...
+Therefore, subsequent query results that include that product will display an accurate size under
+the download link.
+====
+
+=== Workspaces
+
+Each user can create multiple workspaces and assign each of them a descriptive name. Each workspace
+can contain multiple <<Save a Search, saved searches>> and contain multiple saved records
+(metacards). Workspaces are saved for each user and are loaded when the user logs in. Workspaces
+and their contents are persisted, so they survive if {branding} is restarted. Currently, workspaces
```
Comment (Lambeaux) [inline catalog/ui/search-ui/standard/src/main/doc/_contents/help-contents.adoc]: Not sure about comma after `Currently`

File: catalog/ui/search-ui/standard/src/main/doc/_contents/help-contents.adoc (hunk 5 above)
Comment (jrnorth) [inline catalog/ui/search-ui/standard/src/main/doc/_contents/help-contents.adoc]: I think you're right. Also, I think `dropdown` is one word.

**Comment by Lambeaux** [comment]
Left minor grammatical comments. Overall LGTM ... :shipit: after fix!

File: catalog/ui/search-ui/standard/src/main/doc/_contents/help-contents.adoc (hunk 7 above)
Comment (ricklarsen) [inline catalog/ui/search-ui/standard/src/main/doc/_contents/help-contents.adoc]: agreed.

File: catalog/ui/search-ui/standard/src/main/doc/_contents/help-contents.adoc (hunk 8 above)
Comment (ricklarsen) [inline catalog/ui/search-ui/standard/src/main/doc/_contents/help-contents.adoc]: done.

File: catalog/ui/search-ui/standard/src/main/doc/_contents/help-contents.adoc (hunk 9 above)
Comment (ricklarsen) [inline catalog/ui/search-ui/standard/src/main/doc/_contents/help-contents.adoc]: it's a Sentence adverb, that modifies the entire sentence.

File: catalog/ui/search-ui/standard/src/main/doc/_contents/help-contents.adoc (hunk 5 above)
Comment (ricklarsen) [inline catalog/ui/search-ui/standard/src/main/doc/_contents/help-contents.adoc]: It's optional. I like it for extra clarity.

File: catalog/ui/search-ui/standard/src/main/doc/_contents/help-contents.adoc (hunk 6 above)
Comment (ricklarsen) [inline catalog/ui/search-ui/standard/src/main/doc/_contents/help-contents.adoc]: see above.

**Comment by kcwire** [comment]
👍

File: pom.xml [hunk 10]
```diff
@@ -612,6 +612,269 @@
             </build>
         </profile>
         <profile>
+            <id>documentation</id>
```
Comment (shaundmorris) [inline pom.xml]: why did the documentation profile get remove and added back?

**Comment by shaundmorris** [comment]
👍

File: pom.xml [hunk 11]
```diff
@@ -612,6 +612,269 @@
...
+            <activation>
+                <file>
+                    <exists>${basedir}/src/assembly/bin.xml</exists>
```
Comment (shaundmorris) [inline pom.xml]: Will this activate it on any module using the maven assembly plugin?

File: pom.xml (hunk 10 above)
Comment (ricklarsen) [inline pom.xml]: It shouldn't have been added back. It exists solely in `distribution/docs` now

File: pom.xml (hunk 11 above)
Comment (ricklarsen) [inline pom.xml]: It would. removed.
